examples/debug/mod.rs

```diff
@@ -15,44 +15,45 @@
 //! Useful things for debugging noise functions.
 
 extern crate image;
-extern crate num_traits;
 
-use self::num_traits::{Float, NumCast};
 use noise::NoiseFn;
+use std;
 use std::path::Path;
 
-fn cast<T: NumCast, R: NumCast>(val: T) -> R {
-    num_traits::cast(val).unwrap()
-}
+#[inline]
+pub fn clamp<T: PartialOrd>(input: T, min: T, max: T) -> T {
+    assert!(max >= min);
 
-fn clamp<F: Float>(val: F, min: F, max: F) -> F {
-    match () {
-        _ if val < min => min,
-        _ if val > max => max,
-        _ => val,
+    let mut x = input;
+    if x < min {
+        x = min;
+    }
+    if x > max {
+        x = max;
     }
+    x
```

Comment:
I based this on https://github.com/Xaeroxe/rfcs/blob/Xaeroxe-clamp-rfc/text/0000-clamp.md so apparently this is the recommended style.